### PR TITLE
Docs for new git branching workflow

### PR DESCRIPTION
**Pull request by jan-warchol**

### CODE REVIEW

**Review by alicja-gruzdz — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -494,6 +495,24 @@ For immidiete effect make sure to run `source ~/.bashrc`. Also you can manage co
...
+## Git branching workflow
+
+There are two important branches: `main` (representing stable version of the
+project) and `devel` (where most of the work happens; this is the default
+branch).
+
+- New features, and especially breaking changes, should go to the `devel` branch.
+- Bugfixes, especially ones that should be published in a bugfix release,
+  should be merged into the `main` branch.
+  - Start such bugfix branches from `main`, and make PRs to `main`.
```
Comment (a-tylenda) [inline README.md:507]: Will such PR trigger GHA?

File: README.md [hunk 2]
```diff
@@ -494,6 +495,24 @@ For immidiete effect make sure to run `source ~/.bashrc`. Also you can manage co
...
+  - After a bugfix is merged, `main` should be merged into `devel` to get the
```
Comment (a-tylenda) [inline README.md:508]: Who is responsible for merging `develop` into `main` ?

Comment (MichalCz) [inline README.md:508]: @jan-warchol I think we should just automate it.

[the remark this answers is not in the document]
Comment (MichalCz) [reply README.md:508]: It's more regarding when you want the feat to appear... if we're at `1.23.1` and you need the change in `1.23.2` as it's a bugfix of some description then `main`, if `1.24.0` is your target, then `devel`.

Comment (MichalCz) [reply README.md:508]: In general the person who's doing the release to NPM.

**Review by MichalCz — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (MichalCz) [reply README.md:507]: I think it should be done via GHA, not manually.

**Comment by MichalCz** [comment]
@jan-warchol can you review the question and answer them?

File: README.md (hunk 1 above)
Comment (jan-warchol) [reply README.md:507]: PRs to `main` will trigger GHA checks. I don't think we can/should automate making the PRs with bugfixes themselves.

**Review by MichalCz — APPROVED** [review]

**Review by MichalCz — APPROVED** [review]

**Pull request merged**